src/main/scala/mesosphere/marathon/state/Migration.scala

```diff
@@ -300,6 +305,56 @@ class MigrationTo0_13(taskRepository: TaskRepository, store: PersistentStore) {
   } yield ()
 }
 
+/**
+  * Implements the following migration logic:
+  * * Load all apps, the logic in AppDefinition.mergeFromProto will create portDefinitions from the deprecated ports
+  * * Save all apps, the logic in [[AppDefinition.toProto]] will save the new portDefinitions and skip the deprecated
+  *   ports
+  */
+class MigrationTo0_16(groupRepository: GroupRepository, appRepository: AppRepository) {
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
+  def migrate(): Future[Unit] = {
+    log.info("Start 0.16 migration")
+    val rootGroupFuture = groupRepository.rootGroup().map(_.getOrElse(Group.empty))
+
+    for {
+      rootGroup <- rootGroupFuture
+      apps = rootGroup.transitiveApps
```

Comment:
Shouldn't `rootGroup.transitiveApps` and `appRepository.apps()` return the same apps?